imblearn/utils/_validation.py

```diff
@@ -93,7 +93,7 @@ def check_neighbors_object(nn_name, nn_object, additional_neighbor=0):
     """
     if isinstance(nn_object, Integral):
         return NearestNeighbors(n_neighbors=nn_object + additional_neighbor)
-    elif isinstance(nn_object, KNeighborsMixin):
+    elif hasattr(nn_object, 'kneighbors') and hasattr(nn_object, 'kneighbors_graph'):
```

Comment:
It might be nice to abstract this logic into a "private" function, in case it's useful elsewhere. Perhaps something like [this](https://github.com/EpistasisLab/tpot/blob/6448bdb71ba08b4a0447c640d2f05a05e1affc21/tpot/operator_utils.py#L121-L128)?

```python
def _is_selector(estimator):
    selector_attributes = [
        "get_support",
        "transform",
        "inverse_transform",
        "fit_transform",
    ]
    return all(hasattr(estimator, attr) for attr in selector_attributes)
```